clang/include/clang/Basic/AttrDocs.td

```diff
@@ -975,6 +975,15 @@ Query for this feature with ``__has_attribute(diagnose_if)``.
   }];
 }
 
+def DiagnoseSpecializationsDocs : Documentation {
+  let Category = DocCatDecl;
+  let Content = [{
+``clang::diagnose_specializations`` can be appied to class templates which
+should not be specialized by users. This is primarily used to diagnose user
+specializations of standard library type traits.
```

Comment:
"applied to class or variable templates which should not be specialized."

I think "by users" should either be omitted or replaced with "by others"/"elsewhere" here, since this could be used by non-standard-library writers too. It's fine in the next sentence though.

Reply:
Are users of a non-standard-library not users?

Reply:
Oh right, I think I misread it.

You still have the "appied"->"applied" typo btw.